### DISCUSSION TITLE
Using Controller with Button

### DISCUSSION THREAD

**Discussion by kg-lee** [react-hook-form/react-hook-form#8247] [General] [answered]
To my understanding, you could use `Controller` to register any component within react-hook-form. 

One of my form controls is a toggle button (ex. toggle 'A' or 'B') and the issue is that this controlled component is not registering on mount, whereas it is registered once `setValue` is called. Here's what I'm trying to do (using form context):

```    
    const { control, setValue } = useFormContext();

    const onClick = () => {
        setValue('button-field', AorB);
    };

    // render
    <Controller name='button-field' control={control} render={() => 
        <Button onClick={onClick}>
          Toggle
        </Button>} 
    />
```

Is there a way to register this button as a controlled component? A workaround would be using `useEffect` to `setValue` on mount for the `button-field` form value but perhaps it could work with `Controller`.

**Comment by Moshyfawn** [comment, marked as answer]
Take a look at [this example](https://codesandbox.io/s/hungry-gwen-stf7ke?file=%2Fsrc%2FDemo.tsx)

**Reply by mohamedanwer123** [reply]
**thank you** it works fine
